app/javascript/components/custom-search-bar.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextInput, Button } from 'carbon-components-react';
+import { Search32 } from '@carbon/icons-react';
+
+const CustomSearchBar = ({ onSearch }) => {
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleSearch = () => {
+    onSearch(searchQuery);
+  };
+
+  return (
+    <div className="custom-search-bar">
+      <TextInput
+        id="search_input"
+        labelText="Search"
+        placeholder="Enter search term"
+        value={searchQuery}
+        onChange={(e) => setSearchQuery(e.target.value)}
+      />
+      <Button
+        kind="primary"
+        iconDescription="Search"
+        onClick={handleSearch}
```

Comment:
Use the method directly in here and there so we can remove 3 more lines for the function.

`onClick={onSearch(searchQuery)}`